### PR TITLE
Move client packages from npm to bower

### PR DESCRIPTION
**Pull request by jmuharsky**
Between NPM and Bower, Bower is the more "official" repository for client-side libraries, especially Angular stuff, and in particular ui-grid offers no npm installation.

This PR converts the existing packages to Bower and references all non-jsapi scripts locally, but does not use ui-grid with Bower.  The version Explorer uses was never published to Bower, and there are some breaking changes that will be fixed in a separate PR.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "devDependencies": {
+    "angular": "~1.4.1",
+    "angular-mocks": "~1.4.1",
+    "angular-material": "~0.10.0",
+    "angular-ui-router": "~0.2.15",
+    "angular-ui-grid": "~3.0.0-rc.22",
+    "angular-bootstrap": "~0.10.0",
+    "bootstrap-css-only": "~3.2.0",
+    "codemirror": "~5.3.0",
+    "jquery": "~2.1.4"
```
Comment (cmccoy) [inline bower.json]: Are these dev dependencies since they're loaded from a CDN?

Comment (jmuharsky) [inline bower.json]: No more dependencies are loaded from CDN's.  All scripts are now served locally due to flakiness/update issues with CDN particularly on ui-grid and angular-material.

**Comment by cmccoy** [comment]
I get:

```
$ bower install
...
Unable to find a suitable version for angular, please choose one:
    1) angular#>=1.2.16 <=1.3.x which resolved to 1.2.28 and is required by angular-ui-grid#3.0.0-rc.22
    2) angular#>= 1.0.8 which resolved to 1.2.28 and is required by angular-ui-router#0.2.15
    3) angular#1.4.1 which resolved to 1.4.1 and is required by angular-animate#1.4.1, angular-aria#1.4.1, angular-mocks#1.4.1
    4) angular#~1.4.1 which resolved to 1.4.1 and is required by perfkit-explorer-deps
    5) angular#>=1 which resolved to 1.4.1 and is required by angular-bootstrap#0.10.0
    6) angular#^1.3.0 || >1.4.0-beta.0 which resolved to 1.4.1 and is required by angular-material#0.10.0

Prefix the choice with ! to persist it to bower.json
```

**Comment by jmuharsky** [comment]
Added a persisted rule for angular#~1.4.1.  PTAL?

**Comment by cmccoy** [comment]
Thanks. LGTM.

Going forward we should grab closure compiler this way too, as recommended by [`gulp-closure-compiler`](https://www.npmjs.com/package/gulp-closure-compiler).

**Pull request merged**